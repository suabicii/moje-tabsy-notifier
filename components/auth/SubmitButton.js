import React from "react";
import {TouchableOpacity} from "react-native";
import {Text} from "react-native";

function SubmitButton({handleSubmit, loading}) {
    return (
        <TouchableOpacity
            onPress={handleSubmit}
            style={{
                backgroundColor: "#78c2ad",
                borderRadius: 24,
                height: 50,
                justifyContent: "center",
                marginBottom: 20,
                marginHorizontal: 15
            }}
        >
            <Text style={{
                color: "#fff",
                fontSize: 24,
                fontWeight: "300",
                textAlign: "center"
            }}>
                {loading ? 'Proszę czekać...' : 'Zaloguj się'}
            </Text>
        </TouchableOpacity>
    );
}

export default SubmitButton;