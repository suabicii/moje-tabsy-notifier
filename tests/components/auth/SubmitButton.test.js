import React from "react";
import renderer from "react-test-renderer";
import SubmitButton from "../../../components/auth/SubmitButton";

it('should correctly render SubmitButton', () => {
    const handleSubmit = jest.fn();
    const tree = renderer.create(<SubmitButton handleSubmit={handleSubmit}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('should change text value when loading is true', () => {
    const handleSubmit = jest.fn();
    const btnBeforeLoading = renderer.create(
        <SubmitButton handleSubmit={handleSubmit}/>
    ).toJSON();
    const btnAfterLoading = renderer.create(
        <SubmitButton handleSubmit={handleSubmit} loading={true}/>
    ).toJSON();
    expect(btnBeforeLoading.children[0].children)
        .not
        .toEqual(btnAfterLoading.children[0].children);

    //EXPLANATION:
    /*
       console.log(btnBeforeLoading.children);
    * [
        {
          type: 'Text',
          props: { style: [Object] },
          children: [ 'Zaloguj siÄ™' ]
        }
    ]
    * */
});