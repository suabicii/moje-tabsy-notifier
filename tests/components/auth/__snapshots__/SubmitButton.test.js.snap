// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render SubmitButton 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#78c2ad",
      "borderRadius": 24,
      "height": 50,
      "justifyContent": "center",
      "marginBottom": 20,
      "marginHorizontal": 15,
      "opacity": 1,
    }
  }
  testID="submit"
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 24,
        "fontWeight": "300",
        "textAlign": "center",
      }
    }
  >
    Zaloguj siÄ™
  </Text>
</View>
`;
